### DISCUSSION TITLE
Azure login won't work - "Missing or invalid email addresse from profile"

### DISCUSSION THREAD

**Discussion by Kaaly** [requarks/wiki#4135] [Error / Bug Report]
### Describe the bug
After login with Azure authentication, user get "Missing or invalid email addresse from profile"

### Steps to reproduce the behavior:

Set up Azure AD auth according to this doc : https://docs.requarks.io/auth/azure
Enable self-registration
Try to log in with Azure AD

### Additionnal information
Wiki.js version: 2.5.201
Database engine: Postgre

### Solution

Issue came from JWT which do not contains email and preferred_username fields.
On our AD, user email is store in unique_name or upn fields.
So authentification can't find e-mail.

We can fix this issue by changing line below in "/server/modules/authentication/azure/authentification.js" file.
We changed : 
`const usrEmail = _.get(profile, '_json.email', null) || _.get(profile, '_json.preferred_username')`
by
`const usrEmail = _.get(profile, '_json.email', null) || _.get(profile, '_json.preferred_username') || _.get(profile, '_json.unique_name')`

After this changed and restart node server, authentification work fine.

Can you fix this issue in further release so that we can update the application?

Thanks

**Comment by SimonCoutureCSTJ** [comment]
@Kaaly Hey I tried your solution but it didn't work on my side. I'm running it in docker. Did you do anything else?

**Comment by lfernandes00** [comment]
@Kaaly I'm having the same problem.. Any update on this?
